src/components/Products.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { fetchProduct } from "../store/actions/productActions";
+import { titleSelector, categorySelector, priceSelector, ratingSelector, stockSelector, totalSelector } from "../store/selectors/productSelector";
+import { listCategorySelector } from "../store/selectors/categorySelector";
+import Category from "./core/Category";
+import { useInView } from "react-intersection-observer";
+import { fetchCategory } from "../store/actions/categoryAction";
+
+const Products = () => {
+  const dispatch = useDispatch();
+  const [page, setPage] = useState(0);
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const [prevCategory, setPrevCategory] = useState("");
+
+  // Per page/scroll fetch limit = 10-15
+  const limit = 10;
+
+  // Fetching Categories from Categories-only API 
+  const listCategory = useSelector(listCategorySelector);
+
+  const titles = useSelector(titleSelector);
+  const categories = useSelector(categorySelector);
+  const prices = useSelector(priceSelector);
+  const ratings = useSelector(ratingSelector);
+  const stocks = useSelector(stockSelector);
+  const loading = useSelector((state) => state.product.loading);
+  const error = useSelector((state) => state.product.error);
+  const total = useSelector(totalSelector);
+
+  useEffect(() => {
+    dispatch(fetchCategory());
+  }, [dispatch]);
+
+  useEffect(() => {
+    const shouldReplace = selectedCategory !== prevCategory;
+    if (selectedCategory !== prevCategory) {
+      setPage(0);
+      setPrevCategory(selectedCategory);
+    }
+    dispatch(fetchProduct(limit, page * limit, selectedCategory, shouldReplace));
+  }, [dispatch, page, selectedCategory, prevCategory]);
+
+  const handleCategoryClick = (category) => {
```

Comment:
on click of a category, need to add query param as **?category=smartphones** in the url, currently not happening

Reply:
Resolved ✅